### DISCUSSION TITLE
Create custom dynamic values

### DISCUSSION THREAD

**Discussion by fairhat** [directus/directus#3006] [Archive]
in the docs it states that we can use:

```
    $CURRENT_USER — The primary key of the currently authenticated user
    $CURRENT_ROLE — The primary key of the role for the currently authenticated user
    $NOW — The current timestamp
```

is there a way to extend these? Or do we have more than just current user and role?
For example I would like to only allow access to items if the items belong to another collection
This way it could be possible to do multi tenant saas in a single project

Eg: Users, Teams, Roles, Movies
User can only see and edit Movies that are allowed for his Role AND belong to his Team

**Comment by rijkvanzanten** [comment]
There's no way to extend these through extensions. I'd like to support "dynamic values" later on, see: (https://github.com/directus/next/discussions/894).

**Reply by fairhat** [reply]
Could you guide me to the part of the source code? I'd like to patch it myself and possibly create a PR

**Reply by rijkvanzanten** [reply]
Patch it yourself how? You'd like to implement directus/directus#2993?

**Reply by fairhat** [reply]
> 
> 
> Patch it yourself how? You'd like to implement directus/directus#2993?

Yes, exactly, i would like to implement it

**Reply by fairhat** [reply]
I assume it should be somewhere here: https://github.com/directus/next/blob/main/api/src/services/authorization.ts
Is that correct?

**Reply by rijkvanzanten** [reply]
This is the main location: https://github.com/directus/next/blob/cc86ae760503cb3e543a3d31ea2aa5bc53555d5d/api/src/utils/parse-filter.ts#L15-L17

Keep in mind that there's an interesting thought problem here: You don't always have access to the full user / payload, so it's not as easy as just plopping in an extra line here..

**Reply by fairhat** [reply]
Do you have some sort of guide on testing PRs?

**Reply by rijkvanzanten** [reply]
We don't have full (unit) test coverage yet, so for now it's a manual review process
